api/src/opentrons/protocol_engine/state/commands.py

```diff
@@ -317,6 +343,11 @@ def raise_if_stop_requested(self) -> None:
         if self.get_stop_requested():
             raise ProtocolEngineStoppedError("Cannot modify a stopped engine.")
 
+    def raise_if_paused_by_blocking_door(self) -> None:
+        """Raise if the engine is currently paused by an open door."""
+        if self.get_status() == EngineStatus.BLOCKED_BY_OPEN_DOOR:
+            raise RobotDoorOpenError
```

Comment:
I think we want a message on this error so the client has something human readable, too

```suggestion
            raise RobotDoorOpenError("Front door is currently open.")
```